vpp-manager/main.go

```diff
@@ -133,13 +139,44 @@ func makeNewVPPIndex() {
 	VPPgotTimeout[currentVPPIndex] = false
 }
 
+type timeSlice []time.Time
+
+func (s timeSlice) Less(i, j int) bool { return s[i].Before(s[j]) }
+func (s timeSlice) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+func (s timeSlice) Len() int           { return len(s) }
+
+func cleanup_corefiles(corePattern string) error {
+	files := make(map[time.Time]string)
+	var times timeSlice = []time.Time{}
+	dir := corePattern[:strings.LastIndex(corePattern, "/")]
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if path != dir {
+			files[info.ModTime()] = path
+			times = append(times, info.ModTime())
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(files) > maxCoreFiles {
+		sort.Sort(times)
+		for _, time := range times[:len(times)-maxCoreFiles] {
+			os.Remove(files[time])
+		}
+	}
+	return nil
+}
+
 func main() {
 	vppDeadChan = make(chan bool, 1)
 	VPPgotSigCHLD = make(map[int]bool)
 	VPPgotTimeout = make(map[int]bool)
 
 	params := startup.GetVppManagerParams()
 
+	cleanup_corefiles(params.CorePattern)
```

Comment:
The code looks good to me, I just think it would be better to move this towards the end of startup/startup.go, which also sets the core pattern.
The maxCoreFiles definition should go with that file's constants as well.